activiti-core/activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/event/impl/TaskCancelledImpl.java

```diff
@@ -23,16 +23,23 @@
 
 public class TaskCancelledImpl extends RuntimeEventImpl<Task, TaskRuntimeEvent.TaskEvents> implements TaskCancelledEvent {
 
+    private String reason;
+
     public TaskCancelledImpl() {
     }
 
-    public TaskCancelledImpl(Task entity) {
+    public TaskCancelledImpl(Task entity, String reason) {
```

Comment:
Can we remove the empty constructor and let "reason" to be final?

Reply:
true, that constructor is not used anymore

Reply:
good catch about the reason field being final (it should never change once the particular instance of the object is created)